Sally7/S7ConnectionHelpers.cs

```diff
@@ -112,7 +112,7 @@ private static void BuildRequestItem(ref RequestItem requestItem, IDataItem data
             requestItem.VariableType = dataItem.VariableType;
         }
 
-        private static int BuildS7JobRequest(Span<byte> buffer, int parameterLength, int dataLength)
+        private static int BuildS7JobRequest(Span<byte> buffer, ushort parameterLength, ushort dataLength)
         {
             ref var start = ref MemoryMarshal.GetReference(buffer);
             var len = parameterLength + dataLength + 17; // Error omitted
```

Comment:
`len` here will be `int`, so the parameters should remain `int`. See [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunGBLAHYYGAMQgQAFAFdcAC2jDsaBjPlRhwAJQMAvAD4G2BgGoGwANxce9BoOFjJdw8qdbdBo6Ys0AvkA==).

That's why -- from my experience -- it's quite often better to keep the parameters CLS compliant or use `uint` (as it's a native size that cpu registers can understand), and do the cast / type interpretion only where it's needed.
Squeezing out perf unfortunately comes with some traps too.

Reply:
Thanks for this one, makes sense but I didn't do any checking when changing to ushort...